textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -23,7 +23,7 @@
  * @author laishuying
  */
 public class RegexMatcher implements IOperator {
-    private final IPredicate predicate;
+    private final RegexPredicate regexPredicate;
```

Comment:
It's a design question, and I am OK with the implementation of doing casting.  If it's a regexpredicate at run time, the code can throw an exception. I want to hear the opinion of @JavierJia and @sandeepreddy602  .